### ISSUE TITLE
Access to elements inside iframe

### ISSUE THREAD

**Issue by stas-sl** [OnetapInc/chromy#17]
Hi, thanks for the nice library!

It would be even nicer if it will support accessing/clicking elements inside an iframe. Do you know how to do it via Chrome DevTools Protocol?

**Comment by dotneet** [comment]
Hi @stas-sl ,

Thank you for using chromy.

I think that's difficult because of security.
Accessing elements inside an iframe is strongly restricted.

**Comment by stas-sl** [comment]
Maybe there is a way to switch current 'active' frame? Like in Nightwatch/Selenium: `browser.frame(frameIndex)`

**Comment by dotneet** [comment]
Sorry, I misunderstood iframe security.
Currently Chromy doesn't support the easy way to realize that.

But you can use evaluate() to access elements inside iframe like this:

index.html
```html
<body>
  <iframe src="frame.html"></iframe>
</body>
```

frame.html
```html
<body>
  <button onclick="alert('hey!')">btn</button>
</body>
```

```js
let chromy = new Chromy({visible: true})
chromy.chain()
      .goto('http://localhost:8080/index.html')
      .console(console.log)
      .evaluate(() => {
        let frame = document.querySelector('iframe')
        let button = frame.contentDocument.body.querySelector('button')
        console.log(button)
        button.click()
      })
      .sleep(3000)
      .result((r) => console.log(r))
      .end()
      .catch(e => console.error(e))
      .then(_ => chromy.close())
```

**Comment by stas-sl** [comment]
Thanks @dotneet 

But, unfortunately this will not work if iframe src is from another domain

I found a working solution here https://github.com/LucianoGanga/simple-headless-chrome/blob/master/lib/actions.js#L562, where I can pass `frameId` along with `selector`

**Comment by dotneet** [comment]
@stas-sl Wonderful! Thank you for telling me the solution!
I will implement the way to deal elements inside an iframe.

**Comment by dotneet** [comment]
Hi @stas-sl 

I've just published new version(v0.3.0) that support iframe!

Here is example:

```js
const Chromy = require('chromy')
const path = require('path')

let chromy = new Chromy({visible: true})
chromy.chain()
      .goto('file://' + path.join(__dirname, '/pages/index.html'))
      .console(msg => console.log(msg))
      .iframe('iframe', async iframe => {
        await iframe.chain()
                    .click('#btn')
                    .evaluate(_ => {
                      console.log(document.querySelector('h1').innerText)
                    })
                    .end()
      })
      .end()
      .catch(e => console.log(e))
      .then(_ => {
        chromy.close()
      })
```

**Comment by stas-sl** [comment]
Thanks, @dotneet, that was quick!
